benchmark_test.go

```diff
@@ -218,3 +220,88 @@ func TestBenchmarkStatsPerPeer(t *testing.T) {
 	})
 	assert.Equal(t, want, statsServer.Aggregated(), "unexpected stats")
 }
+
+func TestBenchmarkOutput(t *testing.T) {
+	// Testing both plaintext and JSON output
+	tests := []struct {
+		name          string
+		opts          BenchmarkOptions
+		wantOutput    []string
+		notWantOutput []string
+	}{
+		{
+			name: "json",
+			opts: BenchmarkOptions{
+				Format: "json",
+			},
+			wantOutput:    []string{"summary", "benchmarkParameters", "maxRPS", "latencies"},
+			notWantOutput: []string{"Errors", "Benchmark parameters", "Max RPS"},
+		},
+		{
+			name: "plaintext",
```

Comment:
given plaintext or the following csv isn't a format option actually, so they all fall into the default "plaintext". That means, the two test cases are the same, and you have duplicated code

one idea is to use a slice of options in the test struct. so you would have
```
tests := []struct{
   name string
   formats []string 
   /// other fields
} {
  {
   name: "json"
   formats: []string{"json", "JSON", "Json", "jSON"},
  },
   {
   name: "others"
   formats: []string{"plaintext", "csv", "blah"},
  },
}
// and in the test logic, use a loop to loop through the slice of options
for _, tt := range tests {
  for _, format := range tt.formats {
    t.Run(fmt.Sprintf("%s-%s", tt.name, format), func(t *testinglT) {
       // your current test logic
    })
  }
}
```